service/dap/server.go

```diff
@@ -831,83 +832,79 @@ func (s *Session) onLaunchRequest(request *dap.LaunchRequest) {
 			FailedToLaunch, "Failed to launch", fmt.Sprintf("invalid debug configuration - %v", err))
 		return
 	}
+	s.config.log.Debug("parsed launch config: ", prettyPrint(args))
 
-	mode := args.Mode
-	if mode == "" {
-		mode = "debug"
+	if args.Mode == "" {
+		args.Mode = "debug"
 	}
-	if !isValidLaunchMode(mode) {
+	if !isValidLaunchMode(args.Mode) {
 		s.sendShowUserErrorResponse(request.Request, FailedToLaunch, "Failed to launch",
-			fmt.Sprintf("invalid debug configuration - unsupported 'mode' attribute %q", mode))
+			fmt.Sprintf("invalid debug configuration - unsupported 'mode' attribute %q", args.Mode))
 		return
 	}
 
-	program := args.Program
-	if program == "" && mode != "replay" { // Only fail on modes requiring a program
+	if args.Program == "" && args.Mode != "replay" { // Only fail on modes requiring a program
 		s.sendShowUserErrorResponse(request.Request, FailedToLaunch, "Failed to launch",
 			"The program attribute is missing in debug configuration.")
 		return
 	}
-
-	if backend := args.Backend; backend != "" {
-		s.config.Debugger.Backend = backend
-	} else {
-		s.config.Debugger.Backend = "default"
+	args.Program, err = filepath.Abs(args.Program)
```

Comment:
Could this change the meaning of the launch configuration?

Reply:
I don't think so because that's what the underlying backend is supposed to do and that's what we promise to the user - interpreting the program relative to the server's working directory. I just made it explicit, so we can log it and it is clear from the code what is going on.

Reply:
Also, that's what the tests are for. Do you think we are missing something key to verify such things?

Reply:
Hmm, now there is cwd that we could also expand before passing it into the debugger to be expanded there. What do you think?